File: packages/android/connector/src/main/java/app/quiltt/connector/QuilttConnector.kt
package app.quiltt.connector

import android.content.Context

class QuilttConnector(context: Context) {
    private var webView: QuilttConnectorWebView = QuilttConnectorWebView(context)
    private var token: String? = null

    fun authenticate(token: String) {
        this.token = token
        // TODO: Implement this
        // webView.authenticate(token)
    }

    fun connect(
        config: QuilttConnectorConnectConfiguration,
        onEvent: ConnectorSDKOnEventCallback? = null,
        onExit: ConnectorSDKOnEventExitCallback? = null,
        onExitSuccess: ConnectorSDKOnExitSuccessCallback? = null,
        onExitAbort: ConnectorSDKOnExitAbortCallback? = null,
        onExitError: ConnectorSDKOnExitErrorCallback? = null
    ): QuilttConnectorWebView {
        webView.load(token, config, onEvent, onExit, onExitSuccess, onExitAbort, onExitError)
        return webView
    }

    fun reconnect(
        config: QuilttConnectorReconnectConfiguration,
        onEvent: ConnectorSDKOnEventCallback? = null,
        onExit: ConnectorSDKOnEventExitCallback? = null,
        onExitSuccess: ConnectorSDKOnExitSuccessCallback? = null,
        onExitAbort: ConnectorSDKOnExitAbortCallback? = null,
        onExitError: ConnectorSDKOnExitErrorCallback? = null
    ): QuilttConnectorWebView {
        webView.load(token, config, onEvent, onExit, onExitSuccess, onExitAbort, onExitError)
        return webView
    }
}

File: packages/ios/Sources/QuilttConnector/QuilttConnector.swift
// swift-tools-version: 5.9
// The swift-tools-version declares the minimum version of Swift required to build this package.

import Foundation
import WebKit

public class QuilttConnector {
    private var webview: QuilttConnectorWebview?
    private var token: String?
    private var connectorId: String?
    private var connectionId: String?

    public init() {
        webview = QuilttConnectorWebview.init()
    }

    public func authenticate(token: String) {
        self.token = token
    }

    public func connect(
        config: QuilttConnectorConnectConfiguration,
        onEvent: ConnectorSDKOnEventCallback? = nil,
        onExit: ConnectorSDKOnEventExitCallback? = nil,
        onExitSuccess: ConnectorSDKOnExitSuccessCallback? = nil,
        onExitAbort: ConnectorSDKOnExitAbortCallback? = nil,
        onExitError: ConnectorSDKOnExitErrorCallback? = nil
    ) -> WKWebView {
        webview!.load(
            token: self.token,
            config: config,
            onEvent: onEvent,
            onExit: onExit,
            onExitSuccess: onExitSuccess,
            onExitAbort: onExitAbort,
            onExitError: onExitError)
        return webview!
    }

    public func reconnect(
        config: QuilttConnectorReconnectConfiguration,
        onEvent: ConnectorSDKOnEventCallback? = nil,
        onExit: ConnectorSDKOnEventExitCallback? = nil,
        onExitSuccess: ConnectorSDKOnExitSuccessCallback? = nil,
        onExitAbort: ConnectorSDKOnExitAbortCallback? = nil,
        onExitError: ConnectorSDKOnExitErrorCallback? = nil
    ) -> WKWebView {
        webview!.load(
            token: self.token,
            config: config,
            onEvent: onEvent,
            onExit: onExit,
            onExitSuccess: onExitSuccess,
            onExitAbort: onExitAbort,
            onExitError: onExitError)
        return webview!
    }
}
